plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -775,12 +832,16 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
 
             if( !chain.is_building_block()) {
                _unapplied_transactions.add_incoming( trx, api_trx, return_failure_trace, next );
+               trx_tracker.cancel();
                return true;
             }
 
             const auto block_deadline = calculate_block_deadline( chain.pending_block_time() );
             push_result pr = push_transaction( block_deadline, trx, api_trx, return_failure_trace, next );
 
+            if (!pr.failed) {
+               trx_tracker.trx_success();
+            }
```

Comment:
Should we count exhausted transactions as successful?

Reply:
They didn't fail, so don't want to count them as failed. We could count as `other`, but `other` is suppose to be time not spent evaluating trxs. We count add a whole new category, but that seems like it would just muddy the water. I think success is the best option. Time we spent successfully executing transactions; which is true.